lib/index.js

```diff
@@ -202,6 +207,7 @@ export default {
           [relpath, '</dev/null']);
         const execOpts = {
           cwd: dirname(configurationFilePath),
+          uniqueKey: relpath,
```

Comment:
I hadn't even thought about this needing to be file specific in the other `uniqueKey` PR, good call!

However, this doesn't take into account there being more than one provider running at a time for the same file. The instances of `atom-linter` _should_ be isolated from each other, but we might as well tag this with the provider name, something like `` `linter-codeclimate::${relpath}` ``.